jwst/datamodels/schemas/cube.schema.yaml

```diff
@@ -52,4 +52,10 @@ allOf:
       default: 0.0
       ndim: 3
       datatype: float32
+    var_flat:
+      title: Integration-specific variances due to flat field
+      fits_hdu: VAR_FLAT
+      default: 0.0
```

Comment:
Is this going to create the array in all data?

And for my understanding - isn't this a 2D array for CubeModel (since this is not the ifu cube)?

Reply:
No, it will only create the array when accessed.  We don't access it on `__init__` in the datamodel, so will only get created (if it doesn't exist already from `ramp_fit`) in the `flat_field` step itself.

`CubeModel` is the multi-integration cube, so we need variance for every integration, even for flat field, as it depends not just on the input 2D flat ERR but also on the 3D science data array.  See algorithm above.